casepro/backend/junebug.py

```diff
@@ -4,16 +4,54 @@
 from . import BaseBackend
 
 
+class IdentityStore(object):
+    '''Implements required methods for accessing the identity data in the
+    identity store.'''
+    def __init__(self, base_url, auth_token, address_type):
+        '''
+        base_url: the base URL where the identity store is located
+        auth_token: the token that should be used for authorized access
+        address_type: the address type of the addresses that we want
+        '''
+        self.base_url = base_url.rstrip('/')
+        self.address_type = address_type
+        self.session = requests.Session()
+        self.session.headers.update({'Authorization': 'Token %s' % auth_token})
+        self.session.headers.update({'Content-Type': 'application/json'})
+
+    def get_paginated_response(self, url, params={}, **kwargs):
+        '''Get the results of all pages of a response.'''
+        results = []
+        while url is not None:
+            r = self.session.get(url, params=params, **kwargs)
+            data = r.json()
+            results += data.get('results', [])
+            url = data.get('next', None)
+            # params are included in the next url
+            params = {}
+        return results
```

Comment:
A generator would be good, I just thought that it wouldn't be needed, since the list of addresses for a particular identity shouldn't be large (but it is paginated, so we do need to deal with the pagination). I'll add a commit to make everything use generators